src/Lumina/Data/Files/Pcb/PcbResourceFile.cs

```diff
@@ -0,0 +1,197 @@
+                
+                var header = new ResourceNode
+                {
+                    Magic = reader.ReadUInt32(),
+                    Version = reader.ReadUInt32(),
+                    HeaderSkip = reader.ReadUInt32(),
+                    GroupLength = reader.ReadUInt32(),
+                    BoundingBox = Common.BoundingBox.Read( reader )
+                };
+                var numVertFloat16 = reader.ReadUInt16();
+                header.Polygons = new Polygon[reader.ReadUInt16()];
+                var numVertFloat32 = reader.ReadUInt16();
+                
+                header.Vertices = new Common.Vector3[numVertFloat16 + numVertFloat32];
+
+                // Padding
+                reader.BaseStream.Position += 2;
+                
+                
+                if( header.GroupLength != 0 )
+                {
+                    header.Children = new List<ResourceNode>();
+
+                    var finalPosition = initialPosition + header.GroupLength;
+                    while( reader.BaseStream.Position + header.HeaderSkip < finalPosition )
+                    {
+                        header.Children.Add( ResourceNode.ReadWithCount( reader, out var nodesRead ) );
+                        totalNodesRead += nodesRead;
+                    }
+
+                    reader.BaseStream.Position = finalPosition;
+                }
+
+                for( var i = 0; i < numVertFloat32; i++ )
+                {
+                    header.Vertices[ i ] = Common.Vector3.Read( reader );
+                }
+                
+                for( var i = numVertFloat32; i < numVertFloat32 + numVertFloat16; i++ )
+                {
+                    header.Vertices[ i ] = Common.Vector3.Read16( reader );
+                }
+
+                for( var i = 0; i < header.Polygons.Length; i++ )
+                {
+                    header.Polygons[i] = Polygon.Read( reader );
+                }
+                
+                
+                return header;
+            }
+        }
+        
+        public struct Polygon
+        {
+            public byte[] VertexIndex;
+            public ushort Unknown;
+            
+            public static Polygon Read( BinaryReader reader )
+            {
+                var polygon = new Polygon
+                {
+                    VertexIndex = reader.ReadBytes( 3 ),
+                };
+
+                // Padding
+                reader.BaseStream.Position += 2;
+
+                polygon.Unknown = reader.ReadUInt16();
+
+                // Padding
+                reader.BaseStream.Position += 5;
+
+                return polygon;
+            }
+        }
+
+        public PcbResourceHeader Nodes;
+
+        public override void LoadFile()
+        {
+            var streamStart = Reader.BaseStream.Position;
+            var isList = Reader.ReadUInt32() != 0;
+            Reader.BaseStream.Position = streamStart;
+
+            if( isList ) return;
```

Comment:
i would probably throw an exception here instead because this is never a valid case